store/mongo/datastore_mongo.go

```diff
@@ -71,101 +65,96 @@ type DataStoreMongoConfig struct {
 }
 
 type DataStoreMongo struct {
-	session     *mgo.Session
+	client      *mongo.Client
 	automigrate bool
 	multitenant bool
 }
 
-func NewDataStoreMongoWithSession(session *mgo.Session) *DataStoreMongo {
+func NewDataStoreMongoWithClient(client *mongo.Client) *DataStoreMongo {
 	return &DataStoreMongo{
-		session: session,
+		client: client,
 	}
 }
 
 func NewDataStoreMongo(config DataStoreMongoConfig) (*DataStoreMongo, error) {
-	//init master session
-	var err error
-	once.Do(func() {
-
-		var dialInfo *mgo.DialInfo
-		dialInfo, err = mgo.ParseURL(config.ConnectionString)
-		if err != nil {
-			return
-		}
-
-		// Set 10s timeout - same as set by Dial
-		dialInfo.Timeout = 10 * time.Second
-
-		if config.Username != "" {
-			dialInfo.Username = config.Username
-		}
-		if config.Password != "" {
-			dialInfo.Password = config.Password
-		}
-
-		if config.SSL {
-			dialInfo.DialServer = func(addr *mgo.ServerAddr) (net.Conn, error) {
-
-				// Setup TLS
-				tlsConfig := &tls.Config{}
-				tlsConfig.InsecureSkipVerify = config.SSLSkipVerify
+	if !strings.Contains(config.ConnectionString, "://") {
```

Comment:
this way the configuration will be different for different services (it already is since you've introduced that kind of change in the deployments service)
can we create a separate task for changing the configuration and db configuration handling for all backend services (if we agree this is the way to go)?

Reply:
Sure. I have no strong objection to this way of doing it, it's just about personal preference. Let's make it a separate task if we decide to do this, otherwise create a task to revert these changes in `tenantadm` and `deployments`.

Reply:
be careful! you will break some setups. often connection string only is "mongodb" with will never work. thats why it is like that.